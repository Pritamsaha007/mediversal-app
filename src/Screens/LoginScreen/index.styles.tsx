import {StyleSheet} from 'react-native';
import {Fonts} from '../../styles/fonts';

const styles = StyleSheet.create({
  scrollView: {
    flex: 1,
  },
  scrollViewContent: {
    flexGrow: 1,
  },
  safeArea: {
    flex: 1,
    backgroundColor: '#0088B1',
  },
  topSection: {
    alignItems: 'center',
    justifyContent: 'center',
    flex: 1,
  },
  carouselContainer: {
    padding: 32,
  },
  headerTextContainer: {
    alignSelf: 'flex-start',
    paddingVertical: 24,
    paddingLeft: 32,
  },
  welcomeText: {
    color: '#FFFFFF',
    fontSize: 20,
    paddingVertical: 4,
    fontFamily: Fonts.JakartaRegular,
  },
  appNameText: {
    color: '#FFFFFF',
    fontSize: 40,
    lineHeight: 42,
    paddingBottom: 4,
    fontFamily: Fonts.JakartaBold,
  },
  taglineText: {
    color: '#FFFFFF',
    fontSize: 20,
    paddingBottom: 14,
    fontFamily: Fonts.JakartaRegular,
  },
  bottomSection: {
    flex: 1,
    paddingTop: 48,
    paddingHorizontal: 32,
    backgroundColor: '#f8f8f8',
    borderTopLeftRadius: 40,
    borderTopRightRadius: 40,
  },
  createAccountContainer: {
    flexDirection: 'row',
    justifyContent: 'center',
  },
  noAccountText: {
    fontSize: 18,
    color: '#000000',
    fontFamily: Fonts.JakartaRegular,
  },
  createOneText: {
    marginTop: 4,
    color: '#0088B1',
    fontFamily: Fonts.JakartaSemiBold,
    fontSize: 14,
  },
  termsText: {
    fontSize: 12,
    textAlign: 'center',
    color: '#666666',
    marginTop: 40,
    marginBottom: 24,
    fontFamily: Fonts.JakartaRegular,
  },
  termsHighlight: {
    color: '#000000',
    fontFamily: Fonts.JakartaRegular,
  },
});
export default styles;
