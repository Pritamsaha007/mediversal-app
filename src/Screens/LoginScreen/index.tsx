import React, {useState, useMemo, useEffect} from 'react';
import {
  View,
  TouchableOpacity,
  SafeAreaView,
  ScrollView,
  UIManager,
  Platform,
  BackHandler,
  Alert,
  Text,
} from 'react-native';

import ToggleButtons from '../../components/ui/ToggleButton';
import styles from './index.styles';
import EmailSignup from '../../components/auth/EmailSignUp';
import MobileLogin from '../../components/auth/MobileLogIn';
import EmailLogin from '../../components/auth/EmailLogIn';
import Carosel from '../../assests/svgs/Carosel.svg';
if (
  Platform.OS === 'android' &&
  UIManager.setLayoutAnimationEnabledExperimental
) {
  UIManager.setLayoutAnimationEnabledExperimental(true);
}

const LoginScreen = () => {
  const [isMobile, setIsMobile] = useState(true);
  const [isSignup, setIsSignup] = useState(false);

  const headerText = useMemo(
    () => (isSignup ? 'Welcome Aboard!!' : 'Welcome To'),
    [isSignup],
  );
  const subHeaderText = useMemo(
    () => (isSignup ? 'Create Account' : 'Please, Log In.'),
    [isSignup],
  );

  useEffect(() => {
    const handleBackPress = () => {
      if (isSignup) {
        setIsSignup(false);
        setIsMobile(false);
        return true;
      }

      Alert.alert('Exit App', 'Are you sure you want to exit?', [
        {text: 'Cancel', style: 'cancel'},
        {text: 'Exit', onPress: () => BackHandler.exitApp()},
      ]);

      return true;
    };

    const backHandler = BackHandler.addEventListener(
      'hardwareBackPress',
      handleBackPress,
    );

    return () => backHandler.remove();
  }, [isSignup]);

  return (
    <SafeAreaView style={styles.safeArea}>
      <ScrollView
        style={styles.scrollView}
        contentContainerStyle={styles.scrollViewContent}>
        <View style={styles.topSection}>
          {(isMobile || isSignup) && (
            <View style={styles.carouselContainer}>
              <Carosel width={280} height={280} />
            </View>
          )}
          <View style={styles.headerTextContainer}>
            <Text style={styles.welcomeText}>{headerText}</Text>

            {!isSignup ? (
              <>
                <Text style={styles.appNameText}>Mediversal</Text>
                <Text style={styles.taglineText}>
                  Easy Healthcare, In Your Hands
                </Text>
              </>
            ) : (
              <Text style={styles.appNameText}>{subHeaderText}</Text>
            )}
          </View>
        </View>

        <View style={styles.bottomSection}>
          {isSignup ? (
            <EmailSignup />
          ) : (
            <>
              <ToggleButtons isMobile={isMobile} setIsMobile={setIsMobile} />
              {isMobile ? <MobileLogin /> : <EmailLogin />}

              {!isMobile && (
                <View style={styles.createAccountContainer}>
                  <Text style={styles.noAccountText}>
                    Don't have an Account?
                  </Text>
                  <TouchableOpacity
                    onPress={() => {
                      setIsSignup(true);
                      setIsMobile(false);
                    }}>
                    <Text style={styles.createOneText}> Create One</Text>
                  </TouchableOpacity>
                </View>
              )}
              {!isMobile && (
                <Text style={styles.termsText}>
                  By logging in, you agree to our
                  <Text style={styles.termsHighlight}>Terms & Conditions</Text>
                </Text>
              )}
            </>
          )}
        </View>
      </ScrollView>
    </SafeAreaView>
  );
};

export default LoginScreen;
