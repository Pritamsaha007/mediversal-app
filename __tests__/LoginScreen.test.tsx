/**
 *
 * What we are testing here:
 *
 * This test suite verifies the initial rendering of the <LoginScreen /> component.
 * It ensures that the default state (mobile login) loads correctly with all key UI elements.
 *
 * The test uses mocked:
 *   - Child components (MobileLogin, EmailLogin, EmailSignup, ToggleButton)
 *   - SVG asset (Carosel.svg)
 *
 * Covered in this test:
 * 1. Confirms static text elements like headings and subheadings render correctly.
 * 2. Verifies that the MobileLogin component is shown by default.
 * 3. Ensures ToggleButton and SVG are rendered.
 * 4. Checks that email login/signup components and related texts are not visible on initial load.
 */

import React from 'react';
import {render, screen} from '@testing-library/react-native';
import LoginScreen from '../src/Screens/LoginScreen';

jest.mock('../src/components/auth/EmailSignUp', () => {
  const {Text} = require('react-native');
  return () => <Text>EmailSignup Component</Text>;
});
jest.mock('../src/components/auth/MobileLogIn', () => {
  const {Text} = require('react-native');
  return () => <Text>MobileLogin Component</Text>;
});
jest.mock('../src/components/auth/EmailLogIn', () => {
  const {Text} = require('react-native');
  return () => <Text>EmailLogin Component</Text>;
});
jest.mock('../src/components/ui/ToggleButton', () => {
  const {Text} = require('react-native');
  return () => <Text>Toggle Button Component</Text>;
});
jest.mock('../src/assests/svgs/Carosel.svg', () => {
  const {Text} = require('react-native');
  return () => <Text>Carosel SVG</Text>;
});

describe('LoginScreen - Initial Render Check', () => {
  it('renders all expected elements in the initial (mobile login) state', () => {
    render(<LoginScreen />);

    expect(screen.getByText('Welcome To')).toBeVisible();
    expect(screen.getByText('Mediversal')).toBeVisible();
    expect(screen.getByText('Easy Healthcare, In Your Hands')).toBeVisible();

    expect(screen.getByText('MobileLogin Component')).toBeVisible();
    expect(screen.getByText('Toggle Button Component')).toBeVisible();
    expect(screen.getByText('Carosel SVG')).toBeVisible();

    expect(screen.queryByText('EmailLogin Component')).toBeNull();
    expect(screen.queryByText('EmailSignup Component')).toBeNull();
    expect(screen.queryByText("Don't have an Account?")).toBeNull();
    expect(screen.queryByText('Create One')).toBeNull();
    expect(screen.queryByText('By logging in, you agree to our')).toBeNull();
  });
});
