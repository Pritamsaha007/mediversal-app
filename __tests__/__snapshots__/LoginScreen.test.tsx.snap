// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginScreen matches snapshot in its initial (mobile login) state 1`] = `null`;
